pcdsdevices/sim/motor.py

```diff
@@ -0,0 +1,69 @@
+import time
+import threading
+
+from ophyd.device import Device, Component as Cpt
+from ophyd.positioner import SoftPositioner
+from ophyd.signal import AttributeSignal
+from ophyd.sim import SynAxis
+
+from pcdsdevices.mv_interface import FltMvInterface
+
+
+class SynAxisInterface(FltMvInterface, SynAxis):
+    """
+    SynAxis that has our special interface on it.
+
+    This can be used to test when you need the readback_func feature or if you
+    just want your test motor to finish immediately.
+    """
+    def move(self, position, *args, **kwargs):
+        return super().set(position)
+
+
+class SimMotor(FltMvInterface, SoftPositioner, Device):
+    """
+    Simulated motor with our interface attached.
+
+    Unlike the SynAxis built into ophyd, this takes some time to reach the
+    destination. Use this when you need some sort of delay.
+    """
+    user_readback = Cpt(AttributeSignal, 'position')
```

Comment:
You can't keep this the the same and just have this update `self.sim_state['position']`?

Reply:
I don't understand this comment, but this class was much easier to implement as a `SoftPositioner` than a `SynAxis` so that might cause some of the discrepancies here. Rewriting as a `SynAxis` would require a full override of the `move` method iirc.

Reply:
Ah did this review too late at night. I thought this inherited from `SynAxis`. I think this could probably still be implemented as a `SynAxis` with a clever `readback_func`:

https://github.com/NSLS-II/ophyd/blob/f4fdae506ac33b1611471eb7f5a406c3b39e1f0c/ophyd/sim.py#L285

That is a nitpick though...

Reply:
Can `readback_func` be hacked to allow a slow update over time?